haystack/preview/components/embedders/sentence_transformers_document_embedder.py

```diff
@@ -95,7 +97,7 @@ def run(self, documents: List[Document]):
         Embed a list of Documents.
         The embedding of each Document is stored in the `embedding` field of the Document.
         """
-        if not isinstance(documents, list) or not isinstance(documents[0], Document):
+        if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):
```

Comment:
Is there a simple way to write this one? :-)

Reply:
I tend to agree but we had a long discussion about this :smiley::
https://github.com/deepset-ai/haystack/pull/5822#discussion_r1327338941
https://github.com/deepset-ai/haystack/pull/5822#discussion_r1339650311